### DISCUSSION TITLE
NodeJS: For long audio I don't get response from deepgram even when it shows OK on log

### DISCUSSION THREAD

**Discussion by hwanchoi510** [deepgram/community#536] [General help] [answered]
### Which Deepgram product are you using?

Deepgram SDKs, Deepgram API

### Details

Even when the request was finished with OK on usage log, my code doesn't receive any response so it gets stuck on the line.
```
const { result, error } = await deepgram.listen.prerecorded.transcribeUrl(
    {
      url,
    },
    {
      model: 'whisper-large',
      smart_format: true,
      utterances: true,
      detect_language: true,
    }
  );
```
using `"@deepgram/sdk": "3.0.0"`

The request was for a audio that was 2.5+ hours and it took 15+ minutes to finish.

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

POST /v1/listen?model=whisper-large&smart_format=true&utterances=true&detect_language=true

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

fafc500a-5d15-44c2-a113-beb10589722a

### If possible, please attach your code or paste it into the text box.
```
const { result, error } = await deepgram.listen.prerecorded.transcribeUrl(
    {
      url,
    },
    {
      model: 'whisper-large',
      smart_format: true,
      utterances: true,
      detect_language: true,
    }
  );
```

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jjmaldonis** [comment, marked as answer]
Hey @hwanchoi510, it looks like you're submitting extremely long audio (2.5+ hours) to Whisper large. Whisper large requires much more processing than other models, so you will likely need to break the file into 30-60 minute chunks.

You may also want to consider using a callback URL to receive the transcripts. JS is not designed for long running requests, and you'll have better results using a callback + server to receive the transcripts.

**Comment by hwanchoi510** [comment]
@jjmaldonis Thanks for the response. I'll try using a callback to see if it works.

I was wondering why it wouldn't work because previously I even tried up to 5+ hours of audio and it worked okay. 

Maybe it has something to do with the warning in the docs?
